File: apps/mobile/test/pickup_delivery_address_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:modu_repair/features/orders/domain/pickup_delivery_address.dart';

void main() {
  group('resolvePickupDelivery', () {
    test('체크 시 배송지=수거지', () {
      final resolved = resolvePickupDelivery(
        sameAsPickup: true,
        pickupAddress: '대구 수성구 화랑로2길 62',
        pickupAddressDetail: '302',
        pickupZipcode: '42037',
        pickupPhone: '01027239490',
        pickupName: '홍길동',
        deliveryAddress: '',
        deliveryAddressDetail: '',
        deliveryZipcode: '',
        deliveryPhone: '',
        deliveryName: '',
      );

      expect(resolved.deliveryAddress, '대구 수성구 화랑로2길 62');
      expect(resolved.deliveryAddressDetail, '302');
      expect(resolved.deliveryZipcode, '42037');
      expect(resolved.deliveryPhone, '01027239490');
      expect(resolved.customerName, '홍길동');
    });

    test('체크 해제 시 다른 배송지를 유지하고 수거지 연락처는 유지', () {
      final resolved = resolvePickupDelivery(
        sameAsPickup: false,
        pickupAddress: '대구 수성구 화랑로2길 62',
        pickupAddressDetail: '302',
        pickupZipcode: '42037',
        pickupPhone: '01027239490',
        pickupName: '홍길동',
        deliveryAddress: '서울 강남구 테헤란로 1',
        deliveryAddressDetail: '101호',
        deliveryZipcode: '06236',
        deliveryPhone: '01011112222',
        deliveryName: '김수령',
      );

      expect(resolved.pickupAddress, '대구 수성구 화랑로2길 62');
      expect(resolved.pickupPhone, '01027239490');
      expect(resolved.deliveryAddress, '서울 강남구 테헤란로 1');
      expect(resolved.deliveryAddress, isNot(resolved.pickupAddress));
      expect(resolved.deliveryPhone, '01011112222');
      expect(resolved.customerName, '김수령');
      expect(resolved.customerPhone, '01011112222');
    });
  });
}

File: apps/web/lib/pickup-delivery-address.test.ts
import { resolvePickupDeliveryFields } from "./pickup-delivery-address";

function assert(cond: unknown, msg: string) {
  if (!cond) throw new Error(msg);
}

const pickup = {
  pickupAddress: "대구 수성구 화랑로2길 62",
  pickupAddressDetail: "302",
  pickupZipcode: "42037",
};

const same = resolvePickupDeliveryFields({
  sameAsPickup: true,
  ...pickup,
  deliveryAddress: "",
  deliveryAddressDetail: "",
  deliveryZipcode: "",
});
assert(same.deliveryAddress === pickup.pickupAddress, "체크 시 배송지=수거지");
assert(same.deliveryAddressDetail === pickup.pickupAddressDetail, "체크 시 상세도 수거지");
assert(same.deliveryZipcode === pickup.pickupZipcode, "체크 시 우편번호도 수거지");

const different = resolvePickupDeliveryFields({
  sameAsPickup: false,
  ...pickup,
  deliveryAddress: "서울 강남구 테헤란로 1",
  deliveryAddressDetail: "101호",
  deliveryZipcode: "06236",
});
assert(different.pickupAddress === pickup.pickupAddress, "수거지는 그대로");
assert(different.deliveryAddress === "서울 강남구 테헤란로 1", "체크 해제 시 다른 배송지 유지");
assert(different.deliveryAddress !== different.pickupAddress, "수거지와 배송지가 다름");
assert(different.deliveryZipcode === "06236", "배송 우편번호 유지");

console.log("pickup-delivery-address.test.ts passed");
